File: psets/1/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

void print_char(char character, int times);

int main(void)
{
    char space = ' ';
    char hash = '#';
    int height;
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);

    int number_spaces = height - 1;
    int number_hashes = 1;

    while (height > 0)
    {

        print_char(space, number_spaces);
        print_char(hash, number_hashes);
        print_char(space, 2);
        print_char(hash, number_hashes);

        printf("\n");

        number_spaces--;
        number_hashes++;
        height--;
    }
}

void print_char(char character, int times)
{
    for (int i = 0; i < times; i++)
    {
        printf("%c", character);
    }
}

File: psets/6/sentimental-mario-more/mario.py
import cs50


def main():
    height = get_height()
    number_spaces = height - 1
    number_hashes = 1

    while height > 0:
        print(" " * number_spaces, end="")
        print("#" * number_hashes, end="")
        print(" " * 2, end="")
        print("#" * number_hashes)

        number_spaces -= 1
        number_hashes += 1
        height -= 1


def get_height():
    while True:
        height = cs50.get_int("height: ")
        if 1 <= height <= 8:
            return height


main()
